src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticsEndpoint.cs

```diff
@@ -213,7 +244,27 @@ static bool IsInvalidNestingWarningWithinComponent(OmniSharpVSDiagnostic d, Sour
 
                 return !(taghelperNode is null);
             }
+
+            // Ideally this would be solved instead by not emitting the "!" at the HTML backing file,
+            // but we don't currently have a system to accomplish that
+            static bool IsInvalidNestingFromBody(OmniSharpVSDiagnostic d, SourceText sourceText, RazorSyntaxTree syntaxTree)
```

Comment:
Hmmmm, instead of checking Body/HTML would it make sense to turn-off all HTML diagnostics inside of "!"'d elements?

Reply:
`<!div></!div>` is valid Razor though

Reply:
Not inside of an `<html>` tag without being inside of a `<body>` tag (unless I'm missing something basic here).